packages/react-storage/src/components/StorageBrowser/providers/store/location/context.tsx

```diff
@@ -0,0 +1,107 @@
+import React from 'react';
+
+import { createContextUtilities } from '@aws-amplify/ui-react-core';
+import { noop } from '@aws-amplify/ui';
+
+import { assertLocationData } from '../../../validators';
+import { LocationData } from '../../../actions';
+
+export const ERROR_MESSAGE =
+  'Invalid `location` value provided as initial value to `LocationProvider.';
+
+export const DEFAULT_STATE: LocationState = {
+  current: undefined,
+  path: undefined,
+};
+
+export type LocationActionType = {
+  type: 'NAVIGATE';
+  location?: LocationData;
+  path?: string;
+};
+
+/**
+ * Stores current `location` and nested `path` values set on
+ * navigate events.
+ *
+ * @example
+ *`"s3://my-bucket/my/prefix/a-path/a-nested-path/"`
+ *
+ * ```
+ * const state: LocationState = {
+ *   location: {
+ *     bucket: 'my-bucket',
+ *     permisson: 'READWRITE',
+ *     prefix: 'my/prefix/',
+ *     type: 'PREFIX',
+ *   },
+ *   path: 'a-path/a-nested-path/',
+ * }
+ *
+ */
+export interface LocationState {
+  /**
+   * current `location` metadata
+   */
+  current: LocationData | undefined;
+
+  /**
+   * current `location` subpath within a `location`
+   */
+  path: string | undefined;
+}
```

Comment:
After going through the PR thinking we should add `key` to this interface along with `path` to avoid needing to concatenate `current.prefix` and `path` throughout the consumer code. `key` being the entire string following the `bucket`, e.g. `${current.prefix}${path}`